modules/batch-tool/src/main/java/com/devonfw/module/batch/common/cli/AbstractJobSubCommand.java

```diff
@@ -0,0 +1,107 @@
+import java.util.List;
+
+import javax.inject.Inject;
+
+import org.springframework.batch.core.BatchStatus;
+import org.springframework.batch.core.JobExecution;
+import org.springframework.batch.core.JobInstance;
+import org.springframework.batch.core.explore.JobExplorer;
+import org.springframework.batch.core.repository.JobRepository;
+
+abstract class AbstractJobSubCommand {
+
+  @Inject
+  protected JobRepository jobRepository;
+
+  @Inject
+  private JobExplorer jobExplorer;
+
+  private Long getLongIdentifier(String jobIdentifier) {
+
+    try {
+      return new Long(jobIdentifier);
+    } catch (NumberFormatException e) {
+      // Not an ID - must be a name
+      return null;
+    }
+  }
+
+  /**
+   * @param jobIdentifier a job execution id or job name
+   * @param minStatus the highest status to exclude from the result
+   * @return
+   */
+  private List<JobExecution> getJobExecutionsWithStatusGreaterThan(String jobIdentifier, BatchStatus minStatus) {
+
+    Long executionId = getLongIdentifier(jobIdentifier);
+    if (executionId != null) {
+      JobExecution jobExecution = this.jobExplorer.getJobExecution(executionId);
+      if (jobExecution.getStatus().isGreaterThan(minStatus)) {
+        return Arrays.asList(jobExecution);
+      }
+      return Collections.emptyList();
+    }
+
+    int start = 0;
+    int count = 100;
+    List<JobExecution> executions = new ArrayList<JobExecution>();
+    List<JobInstance> lastInstances = this.jobExplorer.getJobInstances(jobIdentifier, start, count);
+
+    while (!lastInstances.isEmpty()) {
+
+      for (JobInstance jobInstance : lastInstances) {
+        List<JobExecution> jobExecutions = this.jobExplorer.getJobExecutions(jobInstance);
+        if (jobExecutions == null || jobExecutions.isEmpty()) {
+          continue;
+        }
+        for (JobExecution jobExecution : jobExecutions) {
+          if (jobExecution.getStatus().isGreaterThan(minStatus)) {
+            executions.add(jobExecution);
+          }
+        }
+      }
+
+      start += count;
+      lastInstances = this.jobExplorer.getJobInstances(jobIdentifier, start, count);
+
+    }
+
+    return executions;
+
+  }
+
+  protected List<JobExecution> getStoppedJobExecutions(String jobIdentifier) {
+
+    List<JobExecution> jobExecutions = getJobExecutionsWithStatusGreaterThan(jobIdentifier, BatchStatus.STARTED);
+    if (jobExecutions.isEmpty()) {
+      return null;
+    }
+    List<JobExecution> result = new ArrayList<JobExecution>();
+    for (JobExecution jobExecution : jobExecutions) {
+      if (jobExecution.getStatus() != BatchStatus.ABANDONED) {
+        result.add(jobExecution);
+      }
+    }
+    return result.isEmpty() ? null : result;
+  }
+
+  protected List<JobExecution> getRunningJobExecutions(String jobIdentifier) {
+
+    List<JobExecution> jobExecutions = getJobExecutionsWithStatusGreaterThan(jobIdentifier, BatchStatus.COMPLETED);
+    if (jobExecutions.isEmpty()) {
+      return null;
```

Comment:
Best practice is to return `Collections.emptyList()` instead of `null` to prevent NPEs. You might want to check your code with SonarLint... Surely nice to have as your code seems to properly handle `null`.

Reply:
I copied the code directly from spring batch.... I think it is better to not change it...

Reply:
Then I hope it is just these 4 lines that you copied as otherwise you infridge the CLA that you have signed... We need to take this serious as an open source project. Please write the code manually. Feel free to get a design inspiration but do not copy&paste exact code into devonfw.

Reply:
I simplified and thereby totally rewrote the code here